### PR TITLE
feat: add gru

### LINKED ISSUE #14698: GRU

### PR DESCRIPTION
**Pull request by notjedi**
Closes #14698

### CODE REVIEW

**Comment by notjedi** [comment]
@paulaehab can you please review?

**Comment by notjedi** [comment]
@paulaehab can you please review?

File: ivy/data_classes/container/layers.py [hunk 1]
```diff
@@ -2436,3 +2436,88 @@ def reduce_window(
             prune_unapplied=prune_unapplied,
```
Comment (paulaehab) [inline ivy/data_classes/container/layers.py:2436]: This code must be added in container/experimental/layers

File: ivy_tests/test_ivy/test_functional/test_nn/test_layers.py [hunk 2]
```diff
@@ -970,3 +970,90 @@ def test_lstm_update(
         bias=bias,
```
Comment (paulaehab) [inline ivy_tests/test_ivy/test_functional/test_nn/test_layers.py:970]: This code also needed to be add in test_experimental

File: ivy/functional/ivy/experimental/layers.py [hunk 3]
```diff
@@ -2480,3 +2480,81 @@ def ifftn(
...
+    x: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+    w: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+    r: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+    b: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+    /,
+    *,
+    initial_h: Optional[Union[ivy.Array, ivy.NativeArray, ivy.Container]] = None,
+) -> Tuple[ivy.Array, ivy.Array]:
+    """
+    Compute an one-layer GRU. This operator is usually supported via some custom
```
Comment (paulaehab) [inline ivy/functional/ivy/experimental/layers.py:2496]: This documentation you took it from ONNX documentation you need to write the documentation in your own according to paramters and special cases

**Review by paulaehab — CHANGES_REQUESTED** [review]
You must look at the forward call for this function of each in the following frameworks (torch,tensorflow,paddle,Jax if founded) and check all arguments and to be sure if there is any special cases look at adding superset docs https://unify.ai/docs/ivy/overview/deep_dive/superset_behaviour.html

**Comment by notjedi** [comment]
need a little help here. so after checking the docs for torch, tensorflow and paddle. there are a few parameters that are not common b/w them. for example, the following params are not present in all of them.

1. dropout
2. seq_length
3. bidirectional
4. bias / use_bias

so should i have to add them to the ivy params and manage them within the function?

**Comment by paulaehab** [comment]
> need a little help here. so after checking the docs for torch, tensorflow and paddle. there are a few parameters that are not common b/w them. for example, the following params are not present in all of them.
> 
> 1. dropout
> 2. seq_length
> 3. bidirectional
> 4. bias / use_bias
> 
> so should i have to add them to the ivy params and manage them within the function?
Hi @notjedi sorry for late reply this PR get under my radar 😄 
yes, you have to add these parameters to ivy and handle them in the function to achieve the superset behavior check this link in doc: https://unify.ai/docs/ivy/overview/deep_dive/superset_behaviour.html

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**